File: src/Goods.Domain/Drivers/Driver.cs
using Goods.Domain.Drivers.Enums;

namespace Goods.Domain.Drivers;

public class Driver(
    Guid id,
    String firstName,
    String secondName,
    String lastName,
    Gender gender,
    LicenseCategory[] driverLicenseCategory,
    DateOnly experience,
    DateOnly birthday,
    Decimal payPerHour,
    Boolean isRemoved)
{
    public Guid Id { get; } = id;
    public String FirstName { get; } = firstName;
    public String SecondName { get; } = secondName;
    public String LastName { get; } = lastName;
    public Gender Gender { get; } = gender;
    public LicenseCategory[] DriverLicenseCategory { get; } = driverLicenseCategory;
    public DateOnly Birthday { get; } = birthday;
    public DateOnly Experience { get; } = experience;
    public Decimal PayPerHour { get; } = payPerHour;
    public Boolean IsRemoved { get; } = isRemoved;
}

File: src/Goods.BackOffice/client-app/src/domain/drivers/driver.ts
import { Page } from "../../tools/types/page";
import { DriverLicenseCategory } from "./enums/driverLicenseCategory";
import { Gender } from "./enums/gender";

export class Driver {
  constructor(
    public readonly id: string,
    public readonly firstName: string,
    public readonly secondName: string,
    public readonly lastName: string,
    public readonly gender: Gender,
    public readonly driverLicenseCategory: DriverLicenseCategory[],
    public readonly birthday: Date,
    public readonly experience: Date,
    public readonly payPerHour: number,
  ) {}
}

export function mapToDriversPage(data: any): Page<Driver> {
  return Page.convert(data, mapToDriver);
}

export function mapToDrivers(data: any[]): Driver[] {
  return data.map(mapToDriver);
}

export function mapToDriver(data: any): Driver {
  const category = data.driverLicenseCategory;
  return new Driver(
    data.id,
    data.firstName,
    data.secondName,
    data.lastName,
    data.gender,
    Array.isArray(category) ? category : category != null ? [category] : [],
    data.birthday,
    data.experience,
    data.payPerHour,
  );
}
